plone/app/content/browser/vocabulary.py

```diff
@@ -233,8 +237,22 @@ def __call__(self):
                             _(safe_unicode(val)),
                             context=self.request
                         )
+
                     else:
                         item[key] = val
+                    if key == 'getMimeIcon':
+                        item[key] = None
+                        if  vocab_item.portal_type =='File':
+                            #get mime type icon url from mimetype registry'
+                            portal_url = api.portal.get().absolute_url()
+                            mtt = api.portal.get_tool(
+                                        name='mimetypes_registry')
+                            if vocab_item.getObject().file.contentType:
```

Comment:
thats why i proposed https://github.com/plone/Products.CMFPlone/issues/1995. I suggest to merge 1995 first. (upgrade changes are interdependent)